src/client/canvas/object/shape.ts

```diff
@@ -0,0 +1,18 @@
+import { observable } from 'mobx'
+import { Appearance } from '../appearance/appearance'
+import { BasicAppearance } from '../appearance/basic_appearance'
+import { Geometry } from '../geometry/geometry'
+
+export class Shape<T> {
```

Comment:
I think you can avoid making this class generic and simply using `Geometry` instead?